pkg/scheduler/framework/plugin/topologyspreadconstraints/state.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package topologyspreadconstraints
+
+import (
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+)
+
+// domainName is the name of a topology domain.
+type domainName string
+
+// count is the number of bindings in a topology domain.
+type count int
+
+// bindingCounter is an interface which allows counting the number of bindings in each
+// topology domain.
+type bindingCounter interface {
```

Comment:
nit, there is no consumer for the bindingCounter & bindingCounterByDomain. I'm not sure how they will be used. suggest to move to the next PR where they're called.